app/models/student.rb

```diff
@@ -156,8 +154,23 @@ def submission_for(code_review)
     submissions.find { |submission| submission.code_review_id == code_review.id }
   end
 
+  def pairs(course=nil)
+    selected_attendance_records = course ? attendance_records.where("date between ? and ?", course.try(:start_date), course.try(:end_date)) : attendance_records
+    (selected_attendance_records.map {|s| s.pairings.pluck(:pair_id)}.flatten).compact.sort
+  end
+
   def pairs_on_day(day)
```

Comment:
Based on the name, I would expect this to return the students, not their IDs.

Reply:
Agreed. I should change this to return students.